docs/bnb-greenfield/for-developers/blob-hub.md

```diff
@@ -0,0 +1,384 @@
+
+### Requirement
+
+Go version above 1.22
+
+### Create a bucket on Greenfield
+
+if you don't have a bucket yet, set up one for blob storage. There are a few ways to create one, below shows examples via [greenfield-go-sdk](https://github.com/bnb-chain/greenfield-go-sdk) and using provided script.
+
+#### Use Dcellar
+Visit https://dcellar.io/ and create a bucket with at least 100G read quota per month.
+
+#### use provided script
+You can use the script, before runinng it, modify the the scripts/.env file(the GRANTEE_BUNDLE_ACCOUNT does not need to modified at this moment):
+
+```shell
+bash scripts/set_up.sh --create_bucket
+```
+
+### Get a Bundler Account
+
+Request a bundle account from the Bundle Service, you need to grant the bundle account permission in next step, so that bundle service
+can create object behave of your account.
+
+```shell
+curl -X POST  https://gnfd-mainnet-bundle.nodereal.io/v1/bundlerAccount/0xf74d8897D8BeafDF4b766E19A62078DE84570656
+
+{"address":"0x4605BFc98E0a5EA63D9D5a4a1Df549732a6963f3"}
+```
+
+### Grant fee and permission to the bundle address for creating bundled objects under the bucket
+
+
+#### use go-sdk
+
+```go
+    bucketActions := []permTypes.ActionType{permTypes.ACTION_CREATE_OBJECT}
+    statements := utils.NewStatement(bucketActions, permTypes.EFFECT_ALLOW, nil, sdktypes.NewStatementOptions{})
+    bundleAgentPrincipal, err := utils.NewPrincipalWithAccount(sdk.MustAccAddressFromHex(bundleAddrToGrant))
+    if err != nil {
+        util.Logger.Fatalf("NewPrincipalWithAccount: %v", err)
+        return
+    }
+    
+    _, err = cli.PutBucketPolicy(ctx, BlobBucketName, BundleAgentPrincipal, []*permTypes.Statement{&statements}, sdktypes.PutPolicyOption{})
+    if err != nil {
+        util.Logger.Fatalf("put policy failed: %v", err)
+        return
+    }
+```
+
+grant allowance to the bundle account
+
+below example shows allowance amount of 1 BNB, considered each object creation gas consumed is 0.000006 BNB, it is approximately for 1666667 objects.
+
+```go
+allowanceAmount := math.NewIntWithDecimal(1, 19)
+_, err = cli.GrantBasicAllowance(ctx, bundleAcct.String(), allowanceAmount, nil, gnfdsdktypes.TxOption{})
+if err != nil {
+    util.Logger.Fatalf("grant fee allowance failed: %v", err)
+}
+```
+
+You can find similar example [here](https://github.com/bnb-chain/greenfield-go-sdk/blob/master/examples/permission.go)
+
+#### Use provided script
+Or you can use the script, replace `GRANTEE_BUNDLE_ACCOUNT` with the addr got from Bundle Service in step 2, and modify the `ALLOWANCE` to your expect amount, so that transacion
+gas will be paid your account:
+
+```shell
+bash scripts/set_up.sh --grant
+```
+
+After above steps are done, you can start running the Blob Syncer Service.
+
+### Build the binaries
```

Comment:
This part is not necessary, user are not coming to build the blob hub

Reply:
only shown in the blobhub repo readme